pytest_bdd/feature.py

```diff
@@ -293,7 +293,11 @@ def __init__(self, basedir, filename, encoding="utf-8", strict_gherkin=True):
                     continue
                 mode = get_step_type(clean_line) or mode
 
-                if not scenario and prev_mode not in (types.BACKGROUND, types.GIVEN) and mode in types.STEP_TYPES:
+                allowed_prev_mode = (types.BACKGROUND, types.GIVEN)
+                if not strict_gherkin:
+                    allowed_prev_mode += (types.WHEN, )
```

Comment:
Khm... Yes it's true, so if user write such tests (with disabled strict gherkin), it means, that he consciously release all these checks. Probably it can be done just for writing some specific features.